Eflumerf/trigger complete message

eflumerf opened the pull request:
This branch introduces a BufferToken message to be used by the DataFlow to indicate availability for processing one additional TriggerDecision.

glehmannmiotto commented:
I find the choice of name "BufferToken" extremely misleading. Those tokens have very little to do with any buffer. This is one of the things I was trying to say: the choice of how many outstanding triggers we want to have is in first place a system choice, it is not a matter of capacity of one or the other application/machine. Something like FreeTriggerRecordToken or FreeTriggerRecordSlot would be clearer. 
I don't see any need for the buffer_token_marker. This is a DF message and it is not part of the raw data.
The message doesn't contain the number of free tokens, which means that you are sending tokens individually and letting the trigger count and do book keeping.
Last, if at some point a TR is not completed, the trigger will never know which trigger decision it was.
All in all, I'm not convinced that the proposed mechanism, as implemented here, is ideal.

eflumerf commented:
I've renamed the class to "TriggerDecisionToken" to better reflect its use in the system. I have also added a `trigger_number` field, which can be filled in by the DataFlow system when the token is being issued as the result of a Trigger Record being completed.

bieryAtFnal reviewed (approved):
The changes look good to me (addition of the TriggerCompleteMessage, but not yet removal of the deprecated TriggerInhibit message).  And, I have run successful tests.
